### PR TITLE
Null check in EDPSimple destructor [9891]

### PR DESCRIPTION
**Pull request by jparisu**
Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Instead of adding all those nullptr checks, I would add the check to `EDPUtils::release_payload_pool`

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3114/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/333/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2782/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3146/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3119/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/338/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2785/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3151/)

File: src/cpp/rtps/builtin/discovery/endpoint/EDPUtils.hpp [hunk 1]
```diff
@@ -64,9 +64,12 @@ class EDPUtils
             const HistoryAttributes& history_attr,
             bool is_reader)
     {
-        PoolConfig pool_cfg = PoolConfig::from_history_attributes(history_attr);
-        pool->release_history(pool_cfg, is_reader);
-        TopicPayloadPoolRegistry::release(pool);
+        if (nullptr != pool)
```
Comment (MiguelCompany) [inline src/cpp/rtps/builtin/discovery/endpoint/EDPUtils.hpp:67]: Shared pointers have a bool operator for a reason, please use it.

```suggestion
        if (pool)
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**